### PR TITLE
Adds support for attaching to a TGW selected by its name

### PR DESCRIPTION
**Pull request by danvaida**
# Adds support for attaching to a TGW selected by its name

## Description
We want to be able to attach a VPC to a Transit Gateway by simply passing the name of the TGW.

## Testing Instructions
See the `examples/satellite` dir.

## How to roll out
N/A

## Notes
N/A

## Demo
N/A

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -5,11 +5,12 @@ resource "aws_ec2_transit_gateway_vpc_attachment" "this" {
   provider           = aws.satellite
   count              = local.create ? 1 : 0
   subnet_ids         = data.aws_subnet_ids.this[0].ids
-  transit_gateway_id = var.transit_gateway_id
+  transit_gateway_id = local.transit_gateway_id
   vpc_id             = data.aws_vpc.this[0].id
 
   # When we create the TGW and the association through RAM in one run, we need
   # this to escape the race condition.
+  #  depends_on = [var.ram_resource_association_id, data.aws_ram_resource_share.this.id]
```
Comment (cytopia) [inline main.tf]: This can be removed or?

File: variables.tf [hunk 2]
```diff
@@ -56,10 +56,17 @@ variable "transit_gateway_id" {
...
 variable "subnet_name_keyword_selector" {
   description = "Keyword matching the name of the subnet(s) for which the routing will be added (i.e. private)"
   type        = string
   default     = "private"
 }
+
+variable "transit_gateway_hub_name" {
+  description = "Name of the Transit Gateway to attach to"
+  type        = string
+  default     = ""
```
Comment (cytopia) [inline variables.tf:71]: Shouldn't we make this mandatory, otherwise it can be forgotten and left empty?

File: variables.tf [hunk 3]
```diff
@@ -56,10 +56,17 @@ variable "transit_gateway_id" {
 variable "ram_resource_association_id" {
   description = "Identifier of the Resource Access Manager Resource Association"
```
Comment (cytopia) [inline variables.tf:57]: Can you clarify what that means?

File: variables.tf (hunk 2 above)
Comment (snovikov) [reply variables.tf:71]: This is optional. Either `ram_resource_association_id` or `transit_gateway_hub_name` is defined.

Comment (danvaida) [reply variables.tf:71]: As the PR name suggests, we are introducing an additional way to establish the attachment of the VPC with the TGW. We still support passing the IDs of the resources, explicitly.
Therefore, we need to allow both options to coexist. We could make use of the TF experimental variable validation feature, to ensure that we don't have _both_ variables set.

File: variables.tf (hunk 3 above)
Comment (danvaida) [reply variables.tf:57]: Sure. Let me know if this does the trick or not: https://aws.amazon.com/blogs/aws/new-aws-resource-access-manager-cross-account-resource-sharing/

TL;DR attempt: you have a shared resource and then you associate this shared resource with something.

**Review by snovikov — APPROVED** [review]
